### PR TITLE
Fix Text Field validation errors not emitting to Screen Readers

### PR DESCRIPTION
**Pull request by cafrias**

### CODE REVIEW

File: packages/react/src/components/BaseInput/types.ts [hunk 1]
```diff
@@ -28,6 +28,13 @@ export interface BaseInputProps extends React.ComponentPropsWithoutRef<'input'>
 	 * of `validateOnInput`.
 	 */
 	validateOnChange?: boolean;
+
+	/**
+	 * Indicates that validation should occur when `onBlur` is triggered. Alias
+	 * of `validateOnBlur`.
+	 */
+	validateOnBlur?: boolean;
```
Comment (sh0ji) [inline packages/react/src/components/BaseInput/types.ts:36]: Could we revert everything having to do with [blur](https://developer.mozilla.org/en-US/docs/Web/API/Element/blur_event)?

Semantically, we want to either trigger validation on [input](https://developer.mozilla.org/en-US/docs/Web/API/Element/input_event) (when the user begins making a change) or on [change](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) (when the value changes).

Practically, `<input type="text" />` triggers `change` and `blur` at the same time (technically `change` happens right before `blur`). But that's not true for all `<input>` types. I put together https://codepen.io/sh0ji/pen/rNRgQaK so you can see what I mean.

Comment (lucas-contreras) [reply packages/react/src/components/BaseInput/types.ts:36]: Yes, this only was a test, I'm gonna revert it ASAP. I understand that `change` and `blur` events are triggered at the same time, however, if the input it's empty is not working neither `change` nor `blur`.

Comment (sh0ji) [reply packages/react/src/components/BaseInput/types.ts]: We don't yet have a "form" pattern, but ultimately that's where input validation should be happening: on [submit](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event). The reality is that people use form elements like inputs for more than just forms, so we need to support input-level validation as well, but I'm okay with leaving it up to applications to decide what to do with empty values.

**Comment by sh0ji** [comment]
@lucas-contreras: aside from the ESLint errors, is there anything else left to do on this?

**Pull request merged**